orchprovider/k8s/v1/util.go

```diff
@@ -193,6 +193,39 @@ func (p *VolumeMarkerBuilder) AddControllerStatuses(pod k8sApiV1.Pod) {
 	_ = p.AddMultiples(string(v1.JivaControllerStatusVK), status, true)
 }
 
+// AddContainerStatuses is to fetch the current state of controller containers
+// inside the pod
+func (p *VolumeMarkerBuilder) AddContainerStatuses(cp k8sApiV1.Pod, volumekey v1.VolumeKey) {
+	for _, current := range cp.Status.ContainerStatuses {
+		value := v1.NilVV
+		if current.State.Waiting != nil {
+			value = v1.ContainerWaitingVV
+		}
+
+		if current.State.Terminated != nil {
+			value = v1.ContainerTerminatedVV
+		}
+
+		if current.State.Running != nil {
+			if current.Ready {
+				value = v1.ContainerRunningVV
+			} else {
+				value = v1.ContainerNotRunningVV
+			}
+		}
+		_ = p.AddMultiples(string(volumekey), string(value), true)
+	}
+}
+
+// IsRunning to compare the state of all containers in a pod
+func (p *VolumeMarkerBuilder) IsRunning(pv *v1.Volume) bool {
```

Comment:
As discussed this logic needs to be changed.
AddMultiples is buggy. please fix it.

Reply:
done

Reply:
why is pv being passed to check if its running. the p instance has all of them.
Nothing needs to be passed for the timebeing.

we can even rename the function as func (p *VolumeMarkerBuilder) IsVolumeRunning() bool {
 // use these keys: ControllerStatusVK, ReplicaStatusVK, ControllerContainerStatusVK, ReplicaContainerStatusVK
 // get the VolumeMarker entity for each key
 // check for the command separated value for each VolumeMarker

}